code/game/objects/items/tcg/tcg.dm

```diff
@@ -313,6 +320,18 @@ GLOBAL_LIST_EMPTY(tcgcard_radial_choices)
 		nu_card.flipped = flipped
 		nu_card.update_icon_state()
 	update_icon_state()
+/**
+ * Signal handler for COMSIG_STORAGE_REMOVED_ITEM. Flips the card to our flipped state, and qdels src if our storage is empty.
+ */
+/obj/item/tcgcard_deck/proc/card_removed_from_storage(datum/storage/tcg/signal_source, obj/item/thing, atom/remove_to_loc, silent)
+	SIGNAL_HANDLER
+
+	var/obj/item/tcgcard/card = thing
+	card.flipped = flipped
+	card.update_appearance(UPDATE_ICON_STATE)
+
+	if(length(signal_source.real_location.contents) == 0)
+		qdel(src)
```

Comment:
Can we just, 
```suggestion
	if(length(contents) == 0)
		qdel(src)
```

Reply:
yeah, but i made it like that in the edge case our storage's real location is NOT us for some reason